controller.go

```diff
@@ -35,205 +32,79 @@ const (
-	defer func() {
-		if err != nil {
-			cancel()
-		}
-	}()
-
-	if _, err = labels.Parse(c.IngressSelector); err != nil {
-		return errors.Wrap(err, "Invalid Ingress selector")
-	}
-
-	c.ingressesIndex, c.ingressesInformer = cache.NewIndexerInformer(
-		cache.NewFilteredListWatchFromClient(
-			c.client.ExtensionsV1beta1().RESTClient(),
-			"ingresses",
-			"",
-			func(options *metaV1.ListOptions) {
-				options.LabelSelector = c.IngressSelector
-				duration := int64(math.MaxInt32)
-				options.TimeoutSeconds = &duration
-			}),
-		&extensionsV1beta1.Ingress{},
-		0,
-		c,
-		cache.Indexers{},
-	)
-
-	go c.ingressesInformer.Run(childCtx.Done())
-
-	if _, err = labels.Parse(c.ConfigMapSelector); err != nil {
-		return errors.Wrap(err, "Invalid ConfigMap selector")
-	}
-
-	c.configMapsIndex, c.configMapsInformer = cache.NewIndexerInformer(
-		cache.NewFilteredListWatchFromClient(
-			c.client.CoreV1().RESTClient(),
-			"configmaps",
-			"",
-			func(options *metaV1.ListOptions) {
-				options.LabelSelector = c.ConfigMapSelector
-				duration := int64(math.MaxInt32)
-				options.TimeoutSeconds = &duration
-			}),
-		&v1.ConfigMap{},
-		0,
-		c,
-		cache.Indexers{},
-	)
-
-	go c.configMapsInformer.Run(childCtx.Done())
-
-	glog.Infoln("Waiting for caches to sync")
-	if !cache.WaitForCacheSync(childCtx.Done(), c.ingressesInformer.HasSynced, c.configMapsInformer.HasSynced) {
-		return fmt.Errorf("could not sync cache")
-	}
-
-	if c.IngressSelector != "" {
-		glog.Infof("Watching Ingress objects matching the following label selector: %v", c.IngressSelector)
-	}
-
-	if c.ConfigMapSelector != "" {
-		glog.Infof("Watching ConfigMap objects matching the following label selector: %v", c.ConfigMapSelector)
-	}
-
-	if len(c.IngressWatchIgnore) > 0 {
-		glog.Infof("Ignoring Ingress objects with the following annotations: %v", c.IngressWatchIgnore)
+		r.Log.Error(err, "could not get ingress object")
+		return ctrl.Result{}, err
 	}
 
-	if len(c.ConfigMapWatchIgnore) > 0 {
-		glog.Infof("Ignoring ConfigMap objects with the following annotations: %v", c.ConfigMapWatchIgnore)
-	}
-
-	c.wakeCh = make(chan struct{}, 1)
-
-	c.Process(childCtx)
-
-	var debounceCh <-chan time.Time
-	for {
-		select {
-		case <-c.wakeCh:
-			if debounceCh == nil {
-				debounceCh = time.After(1 * time.Second)
-			}
-		case <-debounceCh:
-			debounceCh = nil
-			c.Process(childCtx)
-		case <-ctx.Done():
-			return nil
-		}
-	}
-}
-
-func (c *Controller) isIgnored(obj interface{}) bool {
+	ingressClass := ingress.Annotations[IngressClassAnnotation]
 
-	switch object := obj.(type) {
-	case *extensionsV1beta1.Ingress:
-		for _, val := range c.IngressWatchIgnore {
-			if _, exists := object.Annotations[val]; exists {
-				return true
-			}
-		}
-	case *v1.ConfigMap:
-		for _, val := range c.ConfigMapWatchIgnore {
-			if _, exists := object.Annotations[val]; exists {
-				return true
-			}
-		}
-	default:
-		return false
+	if ingressClass != r.IngressClass {
```

Comment:
Now that we're using Ingress under API version networking.k8s.io/v1, we should give precedence to [`Ingress.Spec.IngressClassName`](https://pkg.go.dev/k8s.io/api/networking/v1#IngressSpec) first and then fall back to Ingress class annotation while validating the Ingress class.

Reply:
nice catch @nettoclaudio